libs/sdk-ui-ext/src/internal/components/BaseVisualization.tsx

```diff
@@ -246,13 +252,36 @@ export class BaseVisualization extends React.PureComponent<IBaseVisualizationPro
                 },
                 featureFlags,
                 visualizationProperties: insightProperties(props.insight),
-                renderFun: renderer,
+                renderFun: renderer ?? this.getReactRenderFunction(),
+                unmountFun: unmount ?? this.getReactUnmountFunction(),
             };
 
             this.visualization = visFactory(constructorParams);
         }
     }
 
+    private getReactRenderFunction = () => {
+        if (_createRoot) {
+            return (children: any, element: HTMLElement) => {
+                if (!this.reactRootsMap[element.id]) {
```

Comment:
Pls add invariant check that element.id exist

Reply:
Good point, added.